luxonis_train/attached_modules/losses/fomo_localization_loss.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from typing import Any
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+
+from luxonis_train.enums import TaskType
+from luxonis_train.nodes import FOMOHead
+from luxonis_train.utils import Labels, Packet
+
+from .base_loss import BaseLoss
+
+logger = logging.getLogger(__name__)
+
+
+class FOMOLocalizationLoss(BaseLoss[Tensor, Tensor]):
+    node: FOMOHead
+    supported_tasks: list[tuple[TaskType, ...]] = [
+        (TaskType.BOUNDINGBOX, TaskType.KEYPOINTS)
```

Comment:
Because of easier integration with existing code (mainly metrics e.g. OKS) this model requires both bboxes + keypoint at the center of the box. This is because GT labels in the metrics are automatically pulled from label dict and we then need those to be present.

Reply:
Yeah, certanly a point where we can thing about improving the internal design because this is indeed kind of a hack. You can prepare the predictions for attached modules using wrap() but perhaps you should be also able to access the GT that get's fed into the attached modules. In most cases you would leave as is but for cases like this you could add the "bounding box" GTs at runtime so it is compatible with metrics. But there also might be some problems with defining required labels etc. this way.